src/brogue/Monsters.c

```diff
@@ -2788,26 +2788,44 @@ enum directions scentDirection(creature *monst) {
 // returns true if the resurrection was successful.
 boolean resurrectAlly(const short x, const short y) {
     boolean success;
-    creature *monst = firstCreature(&purgatory);
-    if (monst) {
+    creatureIterator allyIterator = iterateCreatures(&purgatory);
+    creature *monToCheck, *monToRaise;
+    monToCheck = monToRaise = nextCreature(&allyIterator);
+
+    if (!monToCheck) { // No allies dead yet; fail.
+        return false;
+    }
+
+    // Prefer most empowered ally.  In case of tie, prefer ally with greatest monsterID (thus
+    // preferring allies found deeper in the dungeon over ones found higher up and preferring
+    // legendary allies over everyone else).
+    while (monToCheck = nextCreature(&allyIterator)) {
+        if (monToCheck->totalPowerCount > monToRaise->totalPowerCount ||
+            (monToCheck->totalPowerCount == monToRaise->totalPowerCount &&
+             monToCheck->info.monsterID > monToRaise->info.monsterID)) {
+            monToRaise = monToCheck;
```

Comment:
if you make an edit, can you make these agree with CONTRIBUTING.md? Op on left, extra indents or a gap